### ISSUE TITLE
Deployment fails when supplying an empty array for solutions or data sources

### ISSUE THREAD

**Issue by tw3lveparsecs** [tw3lveparsecs/azure-bicep-log-analytics#1]
example below

``` bicep
param deploymentName string = concat('logAnalytics', utcNow())

module logAnalytics './main.bicep' = {
  name: deploymentName
  params: {
    name: 'myLogAnalyticsWorkspace'
    sku: 'PerGB2018'
    retentionInDays: 30
    solutions: []
    dataSources: []
  }
}
```

**Comment by tw3lveparsecs** [comment]
this is a current limitation of Bicep and will be resolved with bicep v0.4 https://github.com/Azure/bicep/issues/1754

**Comment by tw3lveparsecs** [comment]
solved with Azure/bicep#1754
